osu.Game/Rulesets/UI/Scrolling/ScrollingHitObjectContainer.cs

```diff
@@ -18,14 +20,16 @@ public class ScrollingHitObjectContainer : HitObjectContainer
         private readonly IBindable<ScrollingDirection> direction = new Bindable<ScrollingDirection>();
 
         /// <summary>
-        /// Hit objects which require lifetime computation in the next update call.
+        /// A set containing all <see cref="HitObjectContainer.AliveObjects"/> which have an up-to-date layout.
         /// </summary>
-        private readonly HashSet<DrawableHitObject> toComputeLifetime = new HashSet<DrawableHitObject>();
+        private readonly HashSet<DrawableHitObject> layoutComputed = new HashSet<DrawableHitObject>();
 
         /// <summary>
-        /// A set containing all <see cref="HitObjectContainer.AliveObjects"/> which have an up-to-date layout.
+        /// A conservative estimate of maximum bounding box of a <see cref="DrawableHitObject"/>
+        /// with respect to the start time position of the hit object.
+        /// It is used to calculate when the object appears inbound.
         /// </summary>
-        private readonly HashSet<DrawableHitObject> layoutComputed = new HashSet<DrawableHitObject>();
+        protected virtual RectangleF GetDrawRectangle(HitObjectLifetimeEntry entry) => new RectangleF().Inflate(100);
```

Comment:
I don't really like this existing at all. The argument about cyclic dependencies is somewhat sound (given the way this is all set up), but this should really "just work" as it did before...

Assuming 200x200 size will relax lifetime restrictions for some rulesets and possibly lead to a slight performance regression. Not to mention custom rulesets which might suffer even more.

Reply:
This was a workaround for the fact that DHOs added without pooling were late-loaded at the time they became alive. So it was not possible to rely on draw size. But that had other ramifications (as per my other comment below) and so I believe the old code should continue to work as it did before.